gluonnlp/vocab.py

```diff
@@ -456,3 +471,370 @@ def from_json(json_str):
         vocab._bos_token = vocab_dict.get('bos_token')
         vocab._eos_token = vocab_dict.get('eos_token')
         return vocab
+
+
+###############################################################################
+# Subword level vocabulary
+###############################################################################
+class SubwordVocab(object):
+    """Precomputed token index to subword unit mapping.
+
+    Parameters
+    ----------
+    idx_to_token : list of str
+        Known tokens for which the subword units should be precomputed.
+    subword_function
+        Callable to map tokens to lists of subword indices. Can be created with
```

Comment:
should we limit this argument to the available options for now, so that it's easier to serialize?